### PR TITLE
[MIG] Migration of stock_invoice_picking to 8.0

### PR DESCRIPTION
**Pull request by mileo**

### CODE REVIEW

**Comment by eLBati** [comment]
Thanks @mileo 
please see what travis says:

```
./stock_invoice_picking/stock.py:21:1: F401 'fields' imported but unused
./stock_invoice_picking/stock.py:22:1: F401 'RedirectWarning' imported but unused
./stock_invoice_picking/stock.py:22:1: F401 'except_orm' imported but unused
./stock_invoice_picking/stock.py:23:1: F401 'dp' imported but unused
./stock_invoice_picking/stock.py:27:1: E303 too many blank lines (3)
./stock_invoice_picking/stock.py:60:20: W292 no newline at end of file
```

and use `from . import stock` instead of `import stock`

**Comment by mileo** [comment]
@sadamo please see eLBati comment

File: stock_invoice_picking/stock.py [hunk 1]
```diff
@@ -18,30 +18,31 @@
...
 
-    def set_to_be_invoiced(self, cr, uid, ids, context=None):
-        for picking in self.browse(cr, uid, ids, context):
+    @api.multi
+    def set_to_be_invoiced(self):
+        for picking in self:
             if picking.invoice_state == '2binvoiced':
-                raise orm.except_orm(
+                raise Warning(
                     _('Error'),
```
Comment (pedrobaeza) [inline stock_invoice_picking/stock.py]: Warning only takes 1 argument

File: stock_invoice_picking/stock.py [hunk 2]
```diff
@@ -50,26 +51,8 @@ def set_to_be_invoiced(self, cr, uid, ids, context=None):
                 )
             if picking.invoice_state in ('none', 'invoiced'):
                 if picking.invoice_id:
-                    raise orm.except_orm(_('Error'), _(
+                    raise Warning(_('Error'), _(
```
Comment (pedrobaeza) [inline stock_invoice_picking/stock.py]: The same.

**Comment by pedrobaeza** [comment]
Some things to do:
- Extract description into a README.rst file with this template: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst.
- Remove AUTHORS.txt file and put it on the manifest itself and on the README file.
- Include "Odoo Community Association (OCA)" as coauthor for visibility.

About functionality itself, I see there are no prices on pickings, so I suppose you use default product prices. Can you point in any place of the description this case?

**Comment by sadamo** [comment]
@pedrobaeza

Stock_invoice_picking depends indirectly from stock_account that implements the _get_price_unit_invoice (https://github.com/odoo/odoo/blob/8.0/addons/stock_account/stock.py#L160)

That gets price unit from pricelist if partner is given or from product_id.list_price.

My only concern is when user sets type on stock_invoice_onshiping.py then we go to https://github.com/odoo/odoo/blob/8.0/addons/stock_account/stock.py#L119

But stock module always fill move price (https://github.com/odoo/odoo/blob/8.0/addons/stock/stock.py#L1602)

Do you think that we have any other situation that price will not be filled?

**Comment by pedrobaeza** [comment]
Hi, @sadamo, thanks for the explanation. I see that all cases are covered then.

Please satisfy the rest of the comments.

**Comment by eLBati** [comment]
After @pedrobaeza remarks, :+1:

**Comment by pedrobaeza** [comment]
I think we can take the occasion to rename the module to a more meaningful _stock_picking_invoicing_ that also follow common conventions (area + object + function). We can deal with the migration path including the renaming in OpenUpgrade.

File: stock_invoice_picking/__openerp__.py [hunk 3]
```diff
@@ -37,5 +37,5 @@
     'test': [
     ],
     "active": False,
```
Comment (yvaucher) [inline stock_invoice_picking/__openerp__.py]: active key can be removed

File: stock_invoice_picking/stock.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    def set_to_be_invoiced(self):
+        for picking in self:
+            if picking.invoice_state == '2binvoiced':
+                raise Warning(_("Can't update invoice control for picking %s: "
+                                "It's 'to be invoiced' yet") % picking.name)
+            if picking.invoice_state in ('none', 'invoiced'):
+                if picking.invoice_id:
+                    raise Warning(_('Picking %s has linked invoice %s') %
+                                  (picking.name, picking.invoice_id.number))
+                picking.write({'invoice_state': '2binvoiced'})
```
Comment (yvaucher) [inline stock_invoice_picking/stock.py]: For a single field you can write

```
 picking.invoice_state = '2binvoiced'
```

File: stock_invoice_picking/README.rst [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+This module allows to create invoices directly from picking, without having to use sale or purchase orders.
+
+Credits
+=======
+
+Contributors
+------------
+
+Lorenzo Battistini <lorenzo.battistini@agilebg.com>
+
```
Comment (yvaucher) [inline stock_invoice_picking/README.rst]: Don't put line break between contributors

As in template:
https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: stock_invoice_picking/README.rst [hunk 6]
```diff
@@ -0,0 +1,29 @@
+Invoice picking
```
Comment (yvaucher) [inline stock_invoice_picking/README.rst]: Please add License badge as in template: 
https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

**Comment by tafaRU** [comment]
@mileo I addressed @yvaucher's remarks here: https://github.com/kmee/account-invoicing/pull/3 Please have a look at it. Thanks!

**Comment by tafaRU** [comment]
@mileo have you seen kmee#3? Could you please check it, thanks.

**Comment by mileo** [comment]
@tafaRU 

Merged! Sorry for the delay.

Best Regards!

**Comment by yvaucher** [comment]
:+1:

**Comment by tafaRU** [comment]
@mileo NP, thanks for having merged my PR.

:+1:  once travis is green.

**Comment by pedrobaeza** [comment]
Please rename the module to _stock_picking_invoicing_.

**Comment by tafaRU** [comment]
@mileo could you please address https://github.com/OCA/account-invoicing/pull/62#issuecomment-101466805? In this way I could unblock https://github.com/OCA/account-invoicing/pull/74 which depends on this.

**Comment by mileo** [comment]
Sorry for the delay

**Pull request merged**